### ISSUE TITLE
Another problem with regexp

### ISSUE THREAD

**Issue by mingodad** [albertodemichelis/squirrel#142]
Hello !
I just found another problem with regexp, this time it's the missing SQRex initialization to compile a pattern.
See fix in SquiLu here https://github.com/mingodad/squilu/commit/95d0ae570ab70e257ba63419558e0859c06f8b19

The problem can happen when we try to compile an invalid pattern like:
```
local re = regexp("[a");
```
Sometimes we get a runtime error sometimes not depending on what is in memory passing the endo of the string "[a".

On this line:
```
static SQInteger sqstd_rex_class(SQRex *exp)
{
	SQInteger ret = -1;
	SQInteger first = -1,chain;
	if(*exp->_p == SQREX_SYMBOL_BEGINNING_OF_STRING){
		ret = sqstd_rex_newnode(exp,OP_NCLASS);
		exp->_p++;
	}else ret = sqstd_rex_newnode(exp,OP_CLASS);

	if(*exp->_p == _SC(']')) sqstd_rex_error(exp,_SC("empty class"));
	chain = ret;
	while(*exp->_p != _SC(']') && exp->_p != exp->_eol) { //////Here exp->p was NULL

```
It seems that the original intent was to check for the '\0' character at the end of the pattern string and the correct code would be:
```
	while(*exp->_p != _SC(']') && *exp->_p) {
```
As a side note: Why the author do not apply my pull request https://github.com/albertodemichelis/squirrel/pull/128 ?
It's now more than 5 months for a very simple bug fix. Is Squirrel dead ?
Cheers !

**Comment by albertodemichelis** [comment]
I don't apply the pull request #128 because is broken. If scalar is a string, bool or float that will not work. Also, you assume that I want to change squirrels enum behavior. I don't. There are many people that have tons of code written with the current behavior and I can't just change stuff like this without a major release.
Just because I don't merge everything that is thrown my way doesn't mean Squirrel is dead.

**Comment by mingodad** [comment]
Thank you for reply !
And I appreciate that you are still driving squirrel forward.
But why you even didn't care to comment on it in 6 months ? And what about the other bugs reported ?
It would be nice to have at least an answer for then.
Cheers !

**Comment by mingodad** [comment]
Thanks for pointing out my mistake when fixing https://github.com/albertodemichelis/squirrel/pull/128 it's now fixed in SquiLu here https://github.com/mingodad/squilu/commit/1d62218bab7fc495fc88f4c15614353b902a38b1

**Comment by renshijun** [comment]
I use the squirrel to analyze html file. But sqstdrex does not act as posix do like sed grep etc.  I slightly change the behavior of matchnode funtion in sqstdrex.cpp, it act just like what is should be. also I add the lazy match like "<a>.+?</a>".  its very useful for my own needs.

**Comment by renshijun** [comment]
hi, mingodad!
It's chinese lunar new years. We spend these days with my mother. She is 90 years old. When I go back to my work city, I can share my code. 
Do you study the code of julia? The interactive environment of sq or lua is difficult to use. julia or ipython is  good environments. Do you have any idea to solve this problem?

**Comment by mingodad** [comment]
Hello renshijun !
What problem are you talking about ?
```
Do you study the code of julia? The interactive environment of sq or lua is difficult to use. julia or ipython are good environments. Do you have any idea to solve this problem?
```

**Comment by renshijun** [comment]
sq>local a=1

sq>local b=2

sq>print(a+b) 

AN ERROR HAS OCCURED [the index 'a' does not exist]

CALLSTACK
interactive console:1:0 *FUNCTION [main()]

LOCALS
[vargv] ARRAY
[this] TABLE

sq>
----------------------------- above is the interactive ---------------------------
print(a+b) does not remember local a=1 and local b=2.

**Comment by mingodad** [comment]
Yes, that's true, squirrel interactive interpreter parses and interpret only one line at time creating an anonymous function and calling it, so any local variable only exists inside it.
You need to use global variables if you want then to persist.
Cheers !
